src/databricks/sqlalchemy/dialect.py

```diff
@@ -1,52 +1,301 @@
+import os
+
 from databricks import sql
+from databricks import sql as dbsql
+
+import re
+
+from sqlalchemy import types
+
+# we leverage MySQL's implementation of TINYINT and DOUBLE
+from sqlalchemy.types import Integer, BigInteger, SmallInteger, Float, DECIMAL, Boolean
+from sqlalchemy.types import String, DATE, TIMESTAMP
+
+from sqlalchemy import util
+from sqlalchemy import exc
+
+from sqlalchemy.engine import default, interfaces
+from sqlalchemy.sql import compiler
+
 from typing import AnyStr
 
-from sqlalchemy import util, exc, types
-from sqlalchemy.engine import default
 
+# provide a way to debug
+debugbreakpoint = os.getenv("DATABRICKS_DIALECT_DEBUG") or False
 
-class DatabricksDialect(default.DefaultDialect):
```

Comment:
seems `src/databricks/sqlalchemy/dialect.py` already existed before. Where did this come from? 

did we have some sort of support for sqlalchemy dialect previously?

Reply:
The pre-existing ``src/databricks/sqlalchemy/dialect.py`` was a rough scaffolding that Jesse and I put together earlier on (see commit 0378460e0aaa272f6a38193d47c5289cc5d2b98a). I built on this scaffolding.